pkg/resource/deploy/source_eval.go

```diff
@@ -1794,6 +1817,22 @@ func inheritFromParent(child resource.Goal, parent resource.Goal) *resource.Goal
 	if goal.RetainOnDelete == nil {
 		goal.RetainOnDelete = parent.RetainOnDelete
 	}
+	if goal.ResourceHooks == nil {
```

Comment:
Not sure we should be inheriting like this. I thought hooks were like transforms we inherit them without copying them to state?